docs/usage-with-routing.stories.mdx

```diff
@@ -0,0 +1,131 @@
+<Meta title="Documentation/Usage with routing" />
+
+# Using Carbon components with routing libraries
+
+## Contents
+
+- [Usage with React Router v5](#usage-with-react-router-v5)
+- [Usage with React Router v6](#usage-with-react-router-v6)
+- [Supported components](#supported-components)
+
+## Introduction
+
+There is no code in Carbon to implement a specific routing library, we aim to let the user decide which one they want to use and allow some Carbon components to be used as links (listed in the [Supported components](#supported-components) section below).
+
+React Router is one of the most popular routing libraries so we have provided examples below to help with implementing Carbon components as links using both v5 and v6 of this library. There are small but significant differences in the implementation of each version.
+
+## Usage with React Router v5
+
+An example of how to use the Carbon `Button` and `Link` components as links with React Router v5 is shown below, this can be applied to any of the suuported Carbon components.
+Note the use of the `useHistory` hook from `react-router-dom` and the `exact` prop on each `Route`
```

Comment:
```suggestion
Note the use of the `useHistory` hook from `react-router-dom`.
```

The exact prop is only required for the first route and is not unique to this situation.